packages/@react-stately/color/src/useHexColorFieldState.ts

```diff
@@ -0,0 +1,133 @@
+
+export interface HexColorFieldState extends Omit<NumberFieldState, 'value' | 'setValue'> {
+  colorValue: Color,
+  setInputValue: (value: string) => void
+}
+
+export const MIN_COLOR = new Color('#000000');
+export const MAX_COLOR = new Color('#FFFFFF');
+const MIN_COLOR_INT = MIN_COLOR.toHexInt();
+const MAX_COLOR_INT = MAX_COLOR.toHexInt();
+
+export function useHexColorFieldState(
+  props: HexColorFieldProps
+): HexColorFieldState {
+  let {
+    step = 1,
+    value,
+    defaultValue,
+    onChange,
+    validationState
+  } = props;
+  
+  let {color: initialValue} = useColor(value);
+  let {color: initialDefaultValue} = useColor(defaultValue);
+  let [colorValue, setColorValue] = useControlledState<Color>(initialValue, initialDefaultValue, onChange);
+
+  let initialInputValue = (value || defaultValue) && colorValue ? colorValue.toString('hex') : '';
+  let [inputValue, setInputValue] = useState(initialInputValue);
+
+  let increment = useCallback(() => {
+    setColorValue((prevColor: Color) => {
+      let newColor = prevColor;
+      let prevColorInt = prevColor ? prevColor.toHexInt() : MIN_COLOR_INT;
+      let newColorString = prevColor ? prevColor.toString('hex') : '';
+      if (prevColorInt < MAX_COLOR_INT) {
+        newColorString = `#${Math.min(prevColorInt + step, MAX_COLOR_INT).toString(16).padStart(6, '0').toUpperCase()}`;
+        newColor = new Color(newColorString);
+      }
+      setInputValue(newColorString);
+      return newColor;
+    });
+  }, [step, setColorValue, setInputValue]);
+
+  let incrementToMax = useCallback(() => {
+    setColorValue((prevColor: Color) => {
+      let newColor = MAX_COLOR;
+      if (prevColor && prevColor.toHexInt() === MAX_COLOR_INT) {
+        newColor = prevColor;
+      }
+      setInputValue(newColor.toString('hex'));
+      return newColor;
+    });
+  }, [setColorValue, setInputValue]);
+
+  let decrement = useCallback(() => {
+    setColorValue((prevColor: Color) => {
+      let newColor = prevColor ? prevColor : MIN_COLOR;
+      let newColorString = newColor.toString('hex');
+      if (prevColor) {
+        let prevColorInt = prevColor.toHexInt();
+        if (prevColorInt > MIN_COLOR_INT) {
+          newColorString = `#${Math.max(prevColorInt - step, MIN_COLOR_INT).toString(16).padStart(6, '0').toUpperCase()}`;
+          newColor = new Color(newColorString);
+        }
+      }
+      setInputValue(newColorString);
+      return newColor;
+    });
+  }, [step, setColorValue, setInputValue]);
+
+  let decrementToMin = useCallback(() => {
+    setColorValue((prevColor: Color) => {
+      let newColor = MIN_COLOR;
+      if (prevColor && prevColor.toHexInt() === MIN_COLOR_INT) {
+        newColor = prevColor;
+      }
+      setInputValue(newColor.toString('hex'));
+      return newColor;
+    });
+  }, [setColorValue, setInputValue]);
+
+  let setFieldInputValue = (value: string) => {
+    value = value.replace(/[^#0-9a-f]/ig, '');
```

Comment:
I updated the regex to `{0,6}` from `{1,6}` because I was having issues. It wasn't letting me delete (backspace) when there is one character left coz it doesn't match the pattern.